tests/integration/pilot/istioctl_test.go

```diff
@@ -373,6 +373,62 @@ func TestProxyStatus(t *testing.T) {
 		})
 }
 
+// Tests istioctl x proxy-status [<pod>]
+func TestXdsProxyStatus(t *testing.T) {
+	framework.NewTest(t).Features("usability.observability.proxy-status").
+		RequiresSingleCluster().
+		Run(func(ctx framework.TestContext) {
+			istioCtl := istioctl.NewOrFail(ctx, ctx, istioctl.Config{})
+
+			podID, err := getPodID(apps.PodA[0])
+			if err != nil {
+				ctx.Fatalf("Could not get Pod ID: %v", err)
+			}
+
+			var output string
+			var args []string
+			g := gomega.NewWithT(t)
+
+			args = []string{"experimental", "proxy-status"}
+			output, _ = istioCtl.InvokeOrFail(t, args)
+			// Just verify pod A is known to Pilot; implicitly this verifies that
+			// the printing code printed it.
+			g.Expect(output).To(gomega.ContainSubstring(fmt.Sprintf("%s.%s", podID, apps.Namespace.Name())))
+
+			expectSubstrings := func(have string, wants ...string) error {
+				for _, want := range wants {
+					if !strings.Contains(have, want) {
+						return fmt.Errorf("substring %q not found; have %q", want, have)
+					}
+				}
+				return nil
+			}
+
+			retry.UntilSuccessOrFail(t, func() error {
+				args = []string{
+					"experimenta", "proxy-status", fmt.Sprintf("%s.%s", podID, apps.Namespace.Name()),
+				}
+				output, _ = istioCtl.InvokeOrFail(t, args)
+				return expectSubstrings(output, "Clusters Match", "Listeners Match", "Routes Match")
+			})
+
+			// test the --file param
+			retry.UntilSuccessOrFail(t, func() error {
+				filename := "ps-configdump.json"
```

Comment:
use `filepath.Join(t.TempDir(), "ps-configdump.json")` please, or it ends up in the source directory. I sent a PR to fix the existing usage (which I assume you copied this from :slightly_smiling_face: ) already

Reply:
Fixed

Reply:
@howardjohn Please review again.